.github/workflows/build.yml

```diff
@@ -3,38 +3,47 @@ on:
   push:
     branches:
       - master
-    paths-ignore:
```

Comment:
I removed it because I think it's nicer to keep the config simpler, and most importantly Gradle caching will mean that the Gradle tasks will almost instantly run f non-code files are changed.

I tend to err on the side of caution and just run the build as much as possible. Yes, 99.9% of the time updating some files won't change anything, but it's easy to imagine that if the configuration changes (e.g. add some Dokka configuration to include the README in the generated HTML site, or automatically package the LICENSE file into the sources.jar) and forget to un-exclude thee files in the `paths-ignore`.


Also, some files like the LICENSE are probably never going to change so they're basically excluded by default.

I do think that the `.github/**` files shouldn't be excluded. If I make a change to a workflow file, I'd like GitHub to run the updated workflow immediately, at least to check that it's correct!

Anyway, it's your repo so it's up to you :). Just give the word and I'll restore the excludes.

Reply:
I am fine with removing it but do the caches on github really work? I thought it always starts up a new VM in the background and kills it after the workflows are run. That would mean there are no caches sind it always starts "empty". But of course I can be wrong! I am no github expert ;)

Reply:
Maintaining the Gradle Build Cache is what the `gradle/gradle-build-action@v2` will do :). It won't update the cache on PRs, but when this PR is merged to master then it will start caching (and you can check to see the caches in https://github.com/Quillraven/Fleks/actions/caches)

Reply:
Oh very interesting - so there is indeed some caching functionality on a github repository. I am surprised to see that but it is awesome! :)